src/main/java/com/conveyal/datatools/editor/controllers/api/EditorController.java

```diff
@@ -132,6 +133,49 @@ private void registerRoutes() {
         }
     }
 
+    /**
+     * Endpoint to update multiple trips in a single transaction.
+     * Creation of this endpoint was prompted by the need to allow trip ID "flipping",
+     * where trip IDs may swap values in one request. Previously this case would cause an error
+     * since trips were saved one by one, without knowledge that a target trip ID had been "flipped".
+     */
+    private String updateMultipleTrips(Request req, Response res) {
+        /* TODO: Deleting all trips involved in the update is obviously
+            overkill and will have performance impacts. */
+        // FIXME: how does autoCommit in this method affect performance?
+        String namespace = getNamespaceAndValidateSession(req);
+        try {
+            JsonNode jsonNode = mapper.readTree(req.body());
+            JdbcTableWriter tableWriter = new JdbcTableWriter(table, datasource, namespace);
+            JsonNode trips = jsonNode.get("trips");
+            ArrayList<String> returnedJsonStrings = new ArrayList<>();
+
+            if (jsonNode == null) {
+                logMessageAndHalt(req, 400, "JSON body must be provided with trip update request.");
+            }
+            for (final JsonNode trip: trips) {
+                int id = trip.get("id").intValue();
+                // Id == -2 is the front end indication that the trip is being created.
+                if (id != -2){
+                    tableWriter.delete(id, false);
```

Comment:
Could you see if you could use `tableWriter.deleteWhere(...)` to create a single delete transaction?

Reply:
If you do a single transaction, `autocommit = false` shouldn't be necessary.

Reply:
I did take a look at that method, but it's set up to receive a key field and value in the form `key = value`. We would need to refactor it to take `key IN ( ... list of ids we would pass ... )`.